metricbeat/module/system/memory/memory.go

```diff
@@ -29,43 +31,39 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 // Fetch fetches memory metrics from the OS.
 func (m *MetricSet) Fetch() (event common.MapStr, err error) {
-	memStat, err := GetMemory()
+	memStat, err := mem.VirtualMemory()
 	if err != nil {
 		return nil, errors.Wrap(err, "memory")
 	}
-	AddMemPercentage(memStat)
-
-	swapStat, err := GetSwap()
-	if err != nil {
-		return nil, errors.Wrap(err, "swap")
-	}
-	AddSwapPercentage(swapStat)
 
 	memory := common.MapStr{
 		"total": memStat.Total,
 		"used": common.MapStr{
 			"bytes": memStat.Used,
-			"pct":   memStat.UsedPercent,
-		},
-		"free": memStat.Free,
-		"actual": common.MapStr{
-			"free": memStat.ActualFree,
-			"used": common.MapStr{
-				"pct":   memStat.ActualUsedPercent,
-				"bytes": memStat.ActualUsed,
-			},
+			"pct":   system.Round(memStat.UsedPercent/100, .5, 4),
 		},
+		"free":      memStat.Free,
+		"available": memStat.Available,
 	}
 
-	swap := common.MapStr{
-		"total": swapStat.Total,
-		"used": common.MapStr{
-			"bytes": swapStat.Used,
-			"pct":   swapStat.UsedPercent,
-		},
-		"free": swapStat.Free,
+	// OSX /BSD specific
+	if memStat.Active != 0 {
```

Comment:
Would be nice if we could check for something else then `0`. Not sure if this value could be also 0 and be valid?